Documentation/ceph-csi-troubleshooting.md

```diff
@@ -441,3 +441,55 @@ $ rbd ls --id=csi-rbd-node -m=10.111.136.166:6789 --key=AQDpIQhg+v83EhAAgLboWIbl
 ```
 
 Where `-m` is one of the mon endpoints and the `--key` is the key used by the CSI driver for accessing the Ceph cluster.
+
+## Force deleting pods when a Node is lost
+
+When a node is lost, you will see each pod on the node stuck in the `Terminating` state while another pod is rescheduled and is in the `ContainerCreating` state.
+
+We have two workarounds: 
+1. ### Force delete your application pod. After some timeout (8-10 minutes), your pod will be rescheduled
+
+**Solution:**
+
+Delete the pod stuck in `Terminating` state forcefully
+```console
+$ kubectl -n rook-ceph delete pod csi-rbd-demo-depl-69cd495f9b-nl6hf --grace-period 0 --force
+```
+
+**And** then wait for timeout (8-10 minutes).
+
+2. ### To shorten the timeout, blacklist the volume so Rook/Ceph can safely failover the pod sooner.
+
+**Solution:**
+
+The solution is to remove the watcher, following the commands below
+
+```console
+kubectl -n rook-ceph exec deploy/rook-ceph-tools -- bash
+```
+```console
+rbd ls --pool=replicapool
+```
+>```
+> csi-vol-1a1bbb34-7461-11eb-a32b-0a580a81020c
+>```
+
+```console
+rbd status csi-vol-1a1bbb34-7461-11eb-a32b-0a580a81020c --pool=replicapool
+```
+>```
+> Watchers:
+>	watcher=10.130.2.1:0/2076971174 client.14206 cookie=18446462598732840961
+>```
+
+```console
+$ ceph osd blacklist add 10.130.2.1:0/207697117 # to know which watcher to block see above output
+blacklisting 10.130.2.1:0/2076971174 until 2021-02-21T18:12:53.115328+0000 (3600 sec)
+```
+
+After running the above command within a few minutes the pod will be running. **But don't forget to remove the above blacklist watcher**
```

Comment:
At what point should they remove the watcher? When the node is confirmed dead? What if they don't remove the watcher? Does it never get removed automatically?

Reply:
If the node is dead the watcher gets automatically removed. If the control plane is dead the application might be running, in that case, the watcher never gets removed then we need to blacklist it.

Reply:
So we shouldn't tell them to un-blacklist, right? Or else we should be clear when to un-blacklist

Reply:
so when the control plane is dead in this case only we need un-blacklist because in this only we are blocking watcher, right?

Reply:
IIUC from [this comment](https://github.com/rook/rook/pull/7282#pullrequestreview-665380895), the `rbd status` may not be reliable to show a watcher, or at least it doesn't guarantee the node is dead. So it seems like we need to recommend they blacklist a whole node until they can guarantee that node is dead. Otherwise, they risk corruption.

Reply:
Yes, a node might be disconnected from the kubernetes cluster. we can blacklist the node IP in the ceph cluster.